CMakeLists.txt

```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.14)
+project(CTraining)
+
+set(CMAKE_CXX_STANDARD 14)
+
+add_executable(CTraining
+        Exercise_1/main_1.cpp)
```

Comment:
There is no such thing in cpp as "main_1.cpp". Each application has it's own "main.cpp" file.

Reply:
I'd say that the name of file is not important, it could be any .cpp file. Standard (C++14, draft) only says:

> 3.6.1 Main function [basic.start.main]
> 1 A program shall contain a global function called main, which is the designated start of the program. It
> is implementation-defined whether a program in a freestanding environment is required to define a main
> function. [ Note: In a freestanding environment, start-up and termination is implementation-defined; startup
> contains the execution of constructors for objects of namespace scope with static storage duration;
> termination contains the execution of destructors for objects with static storage duration. —end note ]
> 2 An implementation shall not predefine the main function. This function shall not be overloaded. It shall have
> a declared return type of type int, but otherwise its type is implementation-defined. An implementation
> shall allow both
> (2.1) — a function of () returning int and
> (2.2) — a function of (int, pointer to pointer to char) returning int
> as the type of main (8.3.5). In the latter form, for purposes of exposition, the first function parameter is
> called argc and the second function parameter is called argv, where argc shall be the number of arguments
> passed to the program from the environment in which the program is run. If argc is nonzero these arguments
> shall be supplied in argv[0] through argv[argc-1] as pointers to the initial characters of null-terminated
> multibyte strings (ntmbs s) (17.5.2.1.4.2) and argv[0] shall be the pointer to the initial character of a
> ntmbs that represents the name used to invoke the program or "". The value of argc shall be non-negative.
> The value of argv[argc] shall be 0. [ Note: It is recommended that any further (optional) parameters be
> added after argv. —end note ]
> 3 The function main shall not be used within a program. The linkage (3.5) of main is implementation-defined.
> A program that defines main as deleted or that declares main to be inline, static, or constexpr is illformed.
> The name main is not otherwise reserved. [ Example: member functions, classes, and enumerations
> can be called main, as can entities in other namespaces. —end example ]
> 4 Terminating the program without leaving the current block (e.g., by calling the function std::exit(int)
> (18.5)) does not destroy any objects with automatic storage duration (12.4). If std::exit is called to
> end a program during the destruction of an object with static or thread storage duration, the program has
> undefined behavior.
> 5 A return statement in main has the effect of leaving the main function (destroying any objects with automatic
> storage duration) and calling std::exit with the return value as the argument. If control reaches the end
> of main without encountering a return statement, the effect is that of executing
> return 0;

Reply:
Just to clarify what I meant: it is a good practice to use main.cpp for main() function, however it is not enforced anyhow by the C++ standard.